### PR TITLE
Add Client Creds + cert + ADFS automation

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Common/TestCategories.cs [hunk 1]
```diff
@@ -13,5 +13,7 @@ public static class TestCategories
         public const string Selenium = "Selenium";
         public const string LabAccess = "LabAccess";
 
+        public const string ADFS = "ADFS";
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.Common/TestCategories.cs]: How about adding "cloud"

File: tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/ConfidentialClientIntegrationTests.cs [hunk 2]
```diff
@@ -376,8 +376,34 @@ public async Task AcquireTokenWithClientSecretFromAdfsAsync()
...
+                .ExecuteAsync()
+                .ConfigureAwait(false);
+
+            Assert.IsNotNull(authResult);
+            Assert.IsNotNull(authResult.AccessToken);
+            Assert.IsNull(authResult.IdToken);
+        }
+
+        [TestMethod]
+        [TestCategory(TestCategories.ADFS)]
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/ConfidentialClientIntegrationTests.cs]: There are other ADFS tests that do not have a category. should probably be added

File: tests/Microsoft.Identity.Test.Common/TestCategories.cs (hunk 1 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Common/TestCategories.cs]: Do you mean add a separate constant here "Public Cloud" ?

Comment (trwalke) [reply tests/Microsoft.Identity.Test.Common/TestCategories.cs]: yeah, there are other tests that are public cloud tests in the file

File: tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/ConfidentialClientIntegrationTests.cs (hunk 2 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/ConfidentialClientIntegrationTests.cs]: Done.

File: tests/Microsoft.Identity.Test.Common/TestCategories.cs (hunk 1 above)
Comment (trwalke) [reply tests/Microsoft.Identity.Test.Common/TestCategories.cs]: actually, maybe not. ADFS should be fine

File: tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/UsernamePasswordIntegrationTests.cs [hunk 3]
```diff
@@ -74,7 +74,7 @@ public async Task ROPC_ADFSv3Federated_Async()
         }
 
         [TestMethod]
-        [TestCategory("UsernamePasswordIntegrationTests")]
+        [TestCategory(TestCategories.ADFS)]
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/UsernamePasswordIntegrationTests.cs]: yeah, this is good

**Pull request merged**